src/js/angular/guides/steps/complex/ttyg/common/configure-agent/plugin.js

```diff
@@ -0,0 +1,68 @@
+PluginRegistry.add('guide.step', [
+    {
+        guideBlockName: 'configure-agent',
+        getSteps: (options, services) => {
+            const GuideUtils = services.GuideUtils;
+
+            const configureExtractionMethods = (services, options) => {
+                const methods = options.methods || [];
+
+                return methods.map((method) => {
+                    return {
+                        guideBlockName: method.guideBlockName,
+                        options: method.options
+                    };
+                });
+            };
+
+            return [
+                {
+                    guideBlockName: 'wait-for-element-to-show',
+                    options: angular.extend({}, {
+                        elementSelectorToShow: GuideUtils.getElementSelector('.agent-settings-modal'),
+                        timeToWait: 10
+                    }, options)
+                },
+                {
+                    guideBlockName: 'input-element',
+                    options: angular.extend({}, {
+                        content: 'guide.step_plugin.configure-agent.name-input',
+                        class: 'input-agent-name-guide-dialog',
+                        url: '/ttyg',
+                        beforeShowPromise: () => GuideUtils.waitFor(GuideUtils.getGuideElementSelector('agent-form'), 5)
+                            .catch((error) => {
+                                services.toastr.error(services.$translate.instant('guide.unexpected.error.message'));
+                                throw error;
+                            }),
+                        elementSelector: GuideUtils.getGuideElementSelector('agent-name'),
+                        showOn: () => options.name,
```

Comment:
The name of the option is too generic. 'editName' or something like that would be more meaningful.

Reply:
Renamed to `editName`